### DISCUSSION TITLE
Help needed with AfterJobRunsWithError configuration

### DISCUSSION THREAD

**Discussion by lays147** [go-co-op/gocron#795] [Q&A] [answered]
Hi, I have the following job definition:

```go
_, err := scheduler.NewJob(
		gocron.DurationJob(interval),
		gocron.NewTask(func() {
			handler.RunReceivableWindowsWorker(ctx, maxWindowSize)
		}),
		gocron.WithEventListeners(
			gocron.AfterJobRunsWithError(func(jobID uuid.UUID, jobName string, err error) {
				log.Error().Err(err).Msg("Receivable Windows Worker failed execution.")
			}),
		),
		gocron.WithSingletonMode(gocron.LimitModeReschedule),
	)
``` 
I'm wondering why the `AfterJobRunsWithError` is not being triggered, since the invocation of the `handler` is returning an error. Am I missing something? 

Cheers,

**Comment by JohnRoesler** [comment, marked as answer]
Hi @lays147 

You'll need to capture and return the error from the handler, similar to this test https://github.com/go-co-op/gocron/blob/v2/scheduler_test.go#L1778-L1785

Assuming `handler.RunReceivableWindowsWorker(ctx, maxWindowSize)` returns only 1 argument, an error, you could do

```diff
_, err := scheduler.NewJob(
		gocron.DurationJob(interval),
-		gocron.NewTask(func() {
+		gocron.NewTask(func() error {
-			handler.RunReceivableWindowsWorker(ctx, maxWindowSize)
+			return handler.RunReceivableWindowsWorker(ctx, maxWindowSize)
		}),
		gocron.WithEventListeners(
			gocron.AfterJobRunsWithError(func(jobID uuid.UUID, jobName string, err error) {
				log.Error().Err(err).Msg("Receivable Windows Worker failed execution.")
			}),
		),
		gocron.WithSingletonMode(gocron.LimitModeReschedule),
	)
```

**Reply by lays147** [reply]
Hi @JohnRoesler thanks! That worked!
Would you mind updating the docs to reflect this? Because the [example](https://pkg.go.dev/github.com/go-co-op/gocron/v2#AfterJobRunsWithError) doesn't reflect what you've told me.

**Reply by JohnRoesler** [reply]
Feel free to submit a PR!
